apis/cognitoidentityprovider/manualv1alpha1/groupusermembership_types.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manualv1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ExternalAnnotation defines a virtual external name annotation for importing existing resources.
+type ExternalAnnotation struct {
+	Groupname  *string `json:"groupname"`
+	UserPoolID *string `json:"userPoolId"`
+	Username   *string `json:"username"`
+}
+
+// GroupUserMembershipParameters define the desired state of an AWS GroupUserMembership.
+type GroupUserMembershipParameters struct {
+	// Region is which region the Group will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// UserPoolID is the Amazon Cognito Group Name (Group) of the Cognito User-Pool group you want to
+	// add User to.
+	// +immutable
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/cognitoidentityprovider/v1alpha1.UserPool
+	// +crossplane:generate:reference:refFieldName=UserPoolIDRef
+	// +crossplane:generate:reference:selectorFieldName=UserPoolIDSelector
+	UserPoolID string `json:"userPoolId,omitempty"`
+
+	// UserPoolIDRef references to an Group to retrieve its userPoolId
+	// +optional
+	// +immutable
+	UserPoolIDRef *xpv1.Reference `json:"userPoolIdRef,omitempty"`
+
+	// UserPoolIDSelector selects a reference to an Group to retrieve its userPoolId
+	// +optional
+	UserPoolIDSelector *xpv1.Selector `json:"userPoolIdSelector,omitempty"`
+
+	// Groupname is the Amazon Cognito Group Name (Group) of the Cognito User-Pool group you want to
+	// add User to.
+	// +immutable
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/cognitoidentityprovider/v1alpha1.Group
+	// +crossplane:generate:reference:refFieldName=GroupnameRef
```

Comment:
You don't need this - Code Generator will pick it

Reply:
Removed, please review